### PR TITLE
Fix contact search

### LINKED ISSUE #233: contact search erroneous
the current contact search is just a filter about the whole list, done in the ui (the core is not used for searching contacts here)

this is fine for now, however, the search is not incremental as expected, but or-searches for every letter entered. so "hans" searches for all contacts containing "h" "a" "n" or "s"

### PR DESCRIPTION
**Pull request by cyBerta**
fixes #233

### CODE REVIEW

**Comment by r10s** [comment]
i think the contact-list reachable from the tab-bar is ContactListController? if so, we can skip any tweaks for that as this will go away anyway, see https://github.com/deltachat/deltachat-ios/issues/231

**Comment by cyBerta** [comment]
you find the search bar I'm tweaking if you're in the chat list and click on the upper right edit icon.

**Comment by cyBerta** [comment]
relates to #233

File: deltachat-ios/Controller/NewChatViewController.swift [hunk 1]
```diff
@@ -192,6 +192,7 @@ class NewChatViewController: UITableViewController {
                 return cell
             }
         } else if section == 1 {
```
Comment (r10s) [inline deltachat-ios/Controller/NewChatViewController.swift]: when touching code like that, i suggest to switch to use constants as `sectionImport`, `rowScanQr` etc (as eg. in https://github.com/deltachat/deltachat-ios/commit/74f255d6b1ede1b18d1d494e1a6ccf66cb722c60 )

if would not change the whole logic and/or use a better controller base class for now, i would just like to get rid of the numbers.

Comment (cyBerta) [reply deltachat-ios/Controller/NewChatViewController.swift]: I totally agree. In this case section 2 has not always the same meaning though. section 2 can include the "import Devices Contacts" button or the the contact list. That's why I'm not sure how rename section 2 without being super clumsy (like section 2 importOrContactListSection vs. section 3 unimportedDeviceContactListSection). Any suggestions?

Comment (r10s) [reply deltachat-ios/Controller/NewChatViewController.swift]: naming is hard :) i think, the point is not a super-nice-name but that the names make it possible to add sections etc. without breaking things. 

maybe use sth. has `<sectionName>Section` and then inside the section `<sectionName>Section<rowName>Row`, so eg. `unimportedSectionImportRow`. but you will know better :)

**Comment by r10s** [comment]
lgtm + tested

**Pull request merged**